amtl/am-allocator-policies.h

```diff
@@ -49,11 +49,21 @@ class SystemAllocatorPolicy
     }
 
   public:
+#if defined(_DEBUG) && defined(_CRTDBG_MAP_ALLOC)
```

Comment:
Put this inside am_free (same for am_malloc), and then this should be the only place with access to the original malloc/free. There's no need to have all the macros outside this file.